pkg/reconciler/openshift/tektonpipeline/extension.go

```diff
@@ -33,26 +44,60 @@ const (
 	DefaultSA = "pipeline"
 	// DefaultDisableAffinityAssistant is default value of disable affinity assistant flag
 	DefaultDisableAffinityAssistant = "true"
+	DefaultTargetNamespace          = "openshift-pipelines"
 	AnnotationPreserveNS            = "operator.tekton.dev/preserve-namespace"
 	AnnotationPreserveRBSubjectNS   = "operator.tekton.dev/preserve-rb-subject-namespace"
 )
 
 // NoPlatform "generates" a NilExtension
-func OpenShiftExtension(context.Context) common.Extension {
-	return openshiftExtension{}
+func OpenShiftExtension(ctx context.Context) common.Extension {
+	logger := logging.FromContext(ctx)
+	mfclient, err := mfc.NewClient(injection.GetConfig(ctx))
+	if err != nil {
+		logger.Fatalw("error creating client from injected config", zap.Error(err))
+	}
+	mflogger := zapr.NewLogger(logger.Named("manifestival").Desugar())
+	manifest, err := mf.ManifestFrom(mf.Slice{}, mf.UseClient(mfclient), mf.UseLogger(mflogger))
+	if err != nil {
+		logger.Fatalw("error creating initial manifest", zap.Error(err))
+	}
+	ext := openshiftExtension{
+		operatorClientSet: operatorclient.Get(ctx),
+		manifest:          manifest,
+	}
+	return ext
 }
 
-type openshiftExtension struct{}
+type openshiftExtension struct {
+	operatorClientSet versioned.Interface
+	manifest          mf.Manifest
+}
 
 func (oe openshiftExtension) Transformers(comp v1alpha1.TektonComponent) []mf.Transformer {
 	return []mf.Transformer{
 		injectDefaultSA(DefaultSA),
 		setDisableAffinityAssistant(DefaultDisableAffinityAssistant),
+		occommon.ApplyCABundles,
 	}
 }
-func (oe openshiftExtension) PreReconcile(context.Context, v1alpha1.TektonComponent) error {
-	return nil
+func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.TektonComponent) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+
+	// make sure that openshift-pipelines namespace exists
+	namespaceLocation := filepath.Join(koDataDir, "operator-namespace.yaml")
```

Comment:
nit: could you move `operator-namespace.yaml` into itsown directory

Reply:
or handle it programatically using go client as part of prereconcile 🧑‍💻 🤓 
or we can add `ensureTargetNamespace` exists into Pipeline reconciler itself.

I won't call it a blocker to merge this pr.
@concaf could you add an issue for this as a future improvement.

Reply:
Makes sense, created this https://github.com/tektoncd/operator/issues/254

Reply:
there is a way to create resources in extensions eg: https://github.com/tektoncd/operator/blob/6b4b0a872d36eaadb0e02d0a542fdfc303cab586/pkg/reconciler/kubernetes/tektonconfig/extension.go#L49

let us make the issue, `creation of namespace specific`. 
i shall add a comment.

Reply:
> nit: could you move operator-namespace.yaml into itsown directory

done

Reply:
we can merge this.

Note: this will work based on the assumption that targetNamespace will always be `openshiftPipelines`. Infact, it comes from the spec.TargetNamespace of all our CRD instances. 

Ususlly it doesn't change as we create them automatically.

cc @savitaashture @vdemeester